### DISCUSSION TITLE
dynamic contents

### DISCUSSION THREAD

**Discussion by vajrabisj** [40ants/scrapycl#2] [General]
how to use this to catch those dynamic contents? eg, some contents of li under ul cannot be downloaded unless you click the link, etc.

**Comment by svetlyak40wt** [comment]
If you mean sites which use client-side JavaScript to update content, then you have to use something like https://github.com/copyleft/cl-webdriver-client/ instead Scrapycl's `fetch` generic function. Probably a webdriver support might be added to the Scrapycl in future.

**Reply by vajrabisj** [reply]
Thanks. That looks a good library and I will try.

but for my case, when I downloaded the file there is no any li under the ul element. But when I browsed same page online and review through development tools I can see several li under the ul. Don’t know how to scrape such page, it’s a shtml file.

**Reply by svetlyak40wt** [reply]
That is why do you need to use a Webdriver. It downloads the page using a real browser with javascript enabled.

**Reply by vajrabisj** [reply]
Understood, thx!
